### ISSUE TITLE
My  opentherm messages arrive shifted with one hour in the log

### ISSUE THREAD

**Issue by fallen2109** [basschipper/docker-otmonitor#2]
Hi,

Since yesterday I am using your docker otmonitor project and one of the issues I can not fix is that all my messages from the OpenTherm gateway come with timestamp that is exactly one hour earlier than the actual time. This results in an incorrect graph. I am running your source on UBUNTU docker.
This problem does not exist when I run the original otmonitor program (linux 64-bit binary) from the otgw.tclcode.com and I believe the issue is in your branch. 
I will be happy to help in any possible way to fix this issue as otmonitor is quite useful for me.
I am located in the Netherlands - I canbe  reached either via mg.vassilev@quicknet.nl or on my mobile 0650675118.
Thank you for your time!

Kind Regards
Miroslav

**Comment by basschipper** [comment]
Try adding `-e TZ=Europe/Amsterdam` to your `docker run` command.

**Comment by fallen2109** [comment]
Thank you Bas! It is working as expected. One more question - when I was running the standalone version of otmonitor (on a Raspberry Pi) I was using  a parameter "-d /.../otdata.txt" in order to force otmonitor to write to a data log file. Is there a similar parameter that I can pass to the docker version in order to obtain the same functionality from it ?

Because the settings for data file in the otmonitor.conf are not affecting the program at all. This was already like this in the standalone (Linux x64) version, so to force the otmonitor to produce a data log file - I was using the mentioned above parameter.

Again - thank you for your help and time!

Kind regards
Miroslav

**Comment by basschipper** [comment]
Just override the default command:
```
docker run -d -p 8080:8080 \
  -e TZ=Europe/Amsterdam \
  -v $(pwd)/otmonitor:/data \
  basschipper/docker-otmonitor:latest \
  /app/otmonitor --daemon -f /data/otmonitor.conf -d /data/otdata.txt
```

**Comment by fallen2109** [comment]
Thank you again Bas ! This issue is also resolved. One more weird question - I hope I can explain it with the help of the attached picture. On my otmonitor graph screen I see sometime strange dips in the lines of various parameters - I have marked them with red ellipses. It looks as if the otmonitor looses data from time to time and falls to 0 values. As I do not see this kind of behavior when I run the otmonitor on Raspberry Pi 3B - I was wondering if this is may be performance related? I am running docker on Ubuntu 18.04 which is installed on ThinkPad T440s (Core i5-4200U with 8 GB memory and all file systems are on SSD). The only other application on this laptop is the gnome and a UNMS controller (unifi application) which runs in another 7 docker containers and monitors my UNIFI hardware.

Any ideas ?

Again thank you for your time !
![Graph02](https://user-images.githubusercontent.com/45978394/71785495-68868100-3000-11ea-93e6-e1b0d90c6d34.png)

**Comment by basschipper** [comment]
I have no clue... Maybe connectivity issues? To compare, I suggest you check the docker/non-docker version on the same machine.

I don't see this behaviour on my home setup. I'm afraid I can't help you with this issue.

**Comment by fallen2109** [comment]
Hi Bas,

I think after couple of days being hosted non-stop it looks that the program is stabilizing and the issues are gone. It is probably the ubuntu resource sheduler that creates these problems - I did install/remove a lot of packages till I stabilized the laptop to perform properly with it's lid closed as I do need it to run as lean as possible. Now I have a fully fledged PC that costs only 5-10 W an hour and I can logon and use it by just opening the lid. Thank you for your help so far - I am closing this issue and wish you all the best in the future. If you are ever in the vicinity of HOOFDDORP - give me a call and we can have a beer (or two) together! Thanks one more time.
![935B4AF6-BE5E-4CFD-B888-A82223919F8A](https://user-images.githubusercontent.com/45978394/71881942-ad5f0480-3133-11ea-95aa-a67d3fe030d3.png)
